ocaml/typing/primitive.ml

```diff
@@ -293,9 +313,11 @@ let equal_boxed_integer bi1 bi2 =
   | (Pnativeint | Pint32 | Pint64), _ ->
     false
 
-let equal_boxed_vector bi1 bi2 = 
-  match bi1, bi2 with 
-  | Pvec128, Pvec128 -> true 
+let equal_boxed_vector bi1 bi2 =
+  (* For the purposes of layouts/native representations,
+     all 128-bit vector types are equal. *)
+  match bi1, bi2 with
+  | Pvec128 _, Pvec128 _ -> true
```

Comment:
This implementation does not match what I would expect from the name of the function. I do agree that this behavior is correct for the use cases of the function, so I don't think there's a bug. Maybe `equal_boxed_vector_size`?

Reply:
Renamed to `equal_boxed_vector_size`